l2rpn_baselines/utils/BaseDeepQ.py

```diff
@@ -114,32 +114,53 @@ def predict_movement(self, data, epsilon, batch_size=None):
         if batch_size is None:
             batch_size = data.shape[0]
 
-        rand_val = np.random.random(batch_size)
-        q_actions = self._model.predict(data, batch_size=batch_size)
-
-        opt_policy = np.argmax(np.abs(q_actions), axis=-1)
-        opt_policy[rand_val < epsilon] = np.random.randint(0, self._action_size, size=(np.sum(rand_val < epsilon)))
-        return opt_policy, q_actions[0, opt_policy]
+        q_actions = self._model.predict(data, batch_size=batch_size)  # q value of each action
+        opt_policy = np.argmax(q_actions, axis=-1)
+        if epsilon > 0.:
+            rand_val = np.random.random(batch_size)
+            opt_policy[rand_val < epsilon] = np.random.randint(0, self._action_size, size=(np.sum(rand_val < epsilon)))
+        return opt_policy, q_actions[np.arange(batch_size), opt_policy], q_actions
```

Comment:
`q_actions[:, opt_policy]` should suffice

Reply:
I am not sure about that, and the time to compute the "np.arange" is totally neglectible, so i'll keep it as it is :-)